### DISCUSSION TITLE
Optimize game for more niche build cases

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#18042] [Feature requests]
Hi, I am interested in building a gigapark, while I have not been working on one consistently, I have found that gigaparks have notoriously bad framerates and stutter when placing prebuilts, or when you zoom out, or if there's lots of peeps or rides. What is a solution for these issues?
Increase framerate by implementing Vulkan API maybe?
Change prebuilt ride placement code?
Alter game logic calculation

**Comment by ocalhoun6** [comment]
The devs here are always trying to improve optimization, and maybe more optimization and better integration with video drivers might help even more ... but there's only so much you can do before you start running into hardware limitations. It's just a _lot_ of stuff you're asking the computer to keep track of when you make a maximum-size park. Especially when you zoom out and try to see a lot of the park at once. What hardware are you running on?

As for the stuttering when placing prebuilts ... maybe the code for that specifically could be reviewed and optimized more? Sounds like there might be something particularly inefficient going on when showing the ghost sprites for a prebuilt ride, and the hardware stress of already being in such a large park is causing that to be noticeable when it usually isn't. But I wouldn't count on it being possible/feasible to improve.

**Reply by 733737** [reply]
there have been mentions of slow prebuilt ride placement before
